### PR TITLE
test(`reward`): improve tests for `MsgSetRewards`

### PR DESCRIPTION
**Pull request by lubtd**
Part of https://github.com/tendermint/spn/issues/687

- Add tests for `grpc_params` for null queries
- Add test cases for `MsgSetRewards` when setting rewards for a new reward pool
- Add some helpers for test above

### CODE REVIEW

File: testutil/keeper/profile.go [hunk 1]
```diff
@@ -0,0 +1,22 @@
+package keeper
+
+import (
+	"context"
+	"math/rand"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/stretchr/testify/require"
+	"github.com/tendermint/spn/testutil/sample"
+	profiletypes "github.com/tendermint/spn/x/profile/types"
```
Comment (aljo242) [inline testutil/keeper/profile.go:10]:
```suggestion

	"github.com/tendermint/spn/testutil/sample"
	profiletypes "github.com/tendermint/spn/x/profile/types"
```

imports

File: testutil/keeper/profile_test.go [hunk 2]
```diff
@@ -0,0 +1,26 @@
+package keeper_test
+
+import (
+	"testing"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/stretchr/testify/require"
+	testkeeper "github.com/tendermint/spn/testutil/keeper"
+	"github.com/tendermint/spn/testutil/sample"
```
Comment (aljo242) [inline testutil/keeper/profile_test.go:9]:
```suggestion

	testkeeper "github.com/tendermint/spn/testutil/keeper"
	"github.com/tendermint/spn/testutil/sample"
```

imports

**Review by aljo242 — APPROVED** [review]

**Review by giunatale — APPROVED** [review]
